src/main.cpp

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) 2018 DKFZ - ODCF
+ *
+ * Distributed under the MIT License (license terms are at https://github.com/dkfz-odcf/FastqIndEx/blob/master/LICENSE.txt).
+ */
+
+#include "main.h"
+#include "startup/Starter.h"
+
+int main(int argc, const char *argv[]) {
+    Starter *starter = Starter::getInstance();
+    shared_ptr<Runner> runner = starter->createRunner(argc, argv);
+
+    int exitCode = 0;
+    if (!runner->checkPremises()) {
+        cerr << "There were errors preventing FastqIndEx to start:\n";
+        for (auto const &message : runner->getErrorMessages()) {
+            cerr << "\t" << message << "\n";
+        }
+        exitCode = 1;
+    } else {
+        exitCode = runner->run();
+    }
+
+    S3Service::close();
```

Comment:
It is unintuitive to have a separate `S3Service::close()` but no S3Service::open()` (or whatever) in the same lexical scope. Does S3Service call `close()` in the destructor?